src/System.Private.CoreLib/shared/System/Double.cs

```diff
@@ -40,59 +43,86 @@ public struct Double : IComparable, IFormattable, IComparable<Double>, IEquatabl
         public const double NaN = (double)0.0 / (double)0.0;
 
         // 0x8000000000000000 is exactly same as -0.0. We use this explicit definition to avoid the confusion between 0.0 and -0.0.
-        internal static double NegativeZero = Int64BitsToDouble(unchecked((long)0x8000000000000000));
+        internal static double NegativeZero = BitConverter.Int64BitsToDouble(unchecked((long)0x8000000000000000));
```

Comment:
> assuming that Roslyn produces the right set of bits for negative zero.

It does, we have some tests (both in Roslyn and CoreCLR/CoreFX) that depend on this (and that validate the actual bit representation is correct).